ush/run_fv3jedi_exe.py

```diff
@@ -30,67 +30,71 @@ def run_atm_var_analysis(yamlconfig):
     sys.path.append(ufsda_path)
     import ufsda
     # compute config for YAML for executable
-    analysis_subconfig = all_config_dict['analysis options']
-    valid_time = analysis_subconfig['valid_time']
-    h = re.findall('PT(\\d+)H', analysis_subconfig['atm_window_length'])[0]
+    app_mode = all_config_dict['GDASApp mode']
+    if app_mode not in ['hofx', 'variational']:
```

Comment:
I mistakenly specified `GDASApp mode` as `var`.   `run_fv3jedi_exe.py` ran and failed with
```
(gdasapp) Orion-login-1[207] rtreadon$ python run_fv3jedi_exe.py --config sample_var.yaml
2022-04-08 19:13:10:INFO:Loading configuration from sample_var.yaml
2022-04-08 19:13:10:INFO:Created working directory /work/noaa/stmp/rtreadon/gdas_single_test_var
Traceback (most recent call last):
  File "run_fv3jedi_exe.py", line 107, in <module>
    run_fv3jedi_exe(args.config)
  File "run_fv3jedi_exe.py", line 35, in run_fv3jedi_exe
    raise KeyError(f"{app_mode} not supported")
KeyError: 'var not supported'
```

Two comments

1. Can we check `sample.yaml` sooner to catch invalid `GDASApp mode`?   
2. The error message is useful but a user may not know which strings are valid modes.   

This is a not a request to change anything.  We don't want to overly black box everything.  I'm just sharing what I experienced.  It was easy to follow the traceback and see my mistake.

Reply:
Done and done. See https://github.com/NOAA-EMC/GDASApp/pull/44/commits/b764f32fc89679ab7ad8db95011878fe4428d46f

Now it will check before creating the working directory and will write:
`KeyError: "'var' not supported. Current GDASApp modes supported are: hofx | variational"`

Reply:
Works great!  Thank you @CoryMartin-NOAA .   I specified `variationaljunk`  in the input yaml and got

`KeyError: "'variationaljunk' not supported. Current GDASApp modes supported are: hofx | variational"`

Can't be clearer than that!